docker_images/sklearn/tests/generators/generate.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+
+"""Generate artefacts used for testing
+
+Don't run this script directly but use `run.sh` instead.
+
+For the given sklearn version, train models for different task types, upload
+them with and without config to HF Hub, and store their input and predictions
+locally (and in the GH repo).
+
+These artefacts will be used for unit testing the sklearn integration.
+
+"""
+
+import json
+import os
+import pickle
+import sys
+import time
+from pathlib import Path
+from tempfile import mkdtemp, mkstemp
+
+import pandas as pd
+import sklearn
+from huggingface_hub import HfApi
+from huggingface_hub.utils import RepositoryNotFoundError
+from sklearn.datasets import load_iris, make_regression
+from sklearn.ensemble import (
+    HistGradientBoostingClassifier,
+    HistGradientBoostingRegressor,
+)
+from sklearn.linear_model import LinearRegression, LogisticRegression
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from skops import hub_utils
+
+
+SLEEP_BETWEEN_PUSHES = 10
+
+
+def push_repo(repo_name, local_repo):
+    # this token should be allowed to push to the skops-tests org.
+    token = os.environ["SKOPS_TESTS_TOKEN"]
+    repo_id = f"skops-tests/{repo_name}"
+
+    print(f"Pushing {repo_id}")
+
+    client = HfApi()
+    try:
+        client.delete_repo(repo_id, token=token)
+    except RepositoryNotFoundError:
+        # repo does not exist yet
+        pass
+    client.create_repo(repo_id=repo_id, token=token, repo_type="model")
+
+    client.upload_folder(
+        repo_id=repo_id,
+        path_in_repo=".",
+        folder_path=local_repo,
+        commit_message="pushing files to the repo from test generator!",
+        commit_description=None,
+        token=token,
+        repo_type=None,
+        revision=None,
+        create_pr=False,
+    )
+    # prevent AWS "503 Server Error: Slow Down for url" error
+    time.sleep(SLEEP_BETWEEN_PUSHES)
+
+
+def get_tabularregression_data():
+    # Create an artificial regression dataset similar in structure to Iris data
+    n_features = 4
+    X, y = make_regression(150, n_features=n_features, n_informative=2, random_state=42)
+    columns = [f"col{i}" for i in range(n_features)]
+    return pd.DataFrame(X, columns=columns), pd.Series(y)
```

Comment:
Let's use `load_diabetes` instead, more meaningful features and just a single liner to get the data.